src/util/file.c

```diff
@@ -358,6 +358,30 @@ int fileput(char *path, char *string) {
     return(0);
 }
 
+int fileputat(int fd, char *path, char *string) {
```

Comment:
Well, I wanted to be able to run `singularity_file_environment()` at any point of the code. At some points in the code we are chdir'd to different locations, or perhaps chroot'd  to another location. This basically ensures that we're writing into the container no matter where we call this in the program, since the rootfs_fd will remain opened.